packages/js/product-editor/src/utils/register-product-editor-block-type.ts

```diff
@@ -15,12 +14,15 @@ interface BlockRepresentation< T extends Record< string, object > > {
 	settings: Partial< BlockConfiguration< T > >;
 }
 
+// Define a more generic type for the select function to avoid TypeScript errors
+type SelectType = ( store: string ) => Record< string, unknown >;
+
 export function useEvaluationContext( context: Record< string, unknown > ) {
 	const { postType } = context;
 
 	const productId = useEntityId( 'postType', postType );
 
-	const getEvaluationContext = ( select: typeof WPSelect ) => {
+	const getEvaluationContext = ( select: SelectType ) => {
 		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 		// @ts-ignore
```

Comment:
Sorry I missed this part in the first two rounds. Is this something you want to also take care of in this PR? If you remove these ignore comments, you will reveal TS errors on the `select` because it does not have a type for the method `getEditedEntityRecord`

Reply:
I wasn't intending to handle this bit, as my main focus here is the cleanup work in #55399 from the React 18 update. I'm not sure that this part would be related (unless I'm missing something).

That said, it seems a small enough change to where it should be fine to adjust while we're here.

Reply:
Updated in 0ea6d67ded.